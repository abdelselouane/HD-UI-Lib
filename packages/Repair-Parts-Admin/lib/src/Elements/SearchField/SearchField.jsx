import React from 'react';
import PropTypes from 'prop-types';

const SearchField = (props) => {

  const spanStyle = {
    color: '#FF7849',
    fontSize: '20px',
    fontWeight: 500,
    float: 'right',
    padding: '5px',
    position: 'absolute',
    marginTop: '27px',
    cursor: 'pointer',
    textDecoration: 'none',
    marginLeft: '-80px'
  };

  const {
    input,
    action
  } = props;

  return (
    <div className="text-input-container">
      <input
        {...input}
      />
      <a 
        style={spanStyle} 
        id={action.id} 
        onClick={action.onClick}
      >
      {action.text}
      </a>
    </div>
  );
};

SearchField.propTypes = {
  action: PropTypes.shape({
    text: PropTypes.string
  }),
  input: PropTypes.shape({
    id: PropTypes.string,
    type: PropTypes.string,
    placeholder: PropTypes.string,
    value: PropTypes.string,
    onChange: PropTypes.func,
  })
};

export default SearchField;
