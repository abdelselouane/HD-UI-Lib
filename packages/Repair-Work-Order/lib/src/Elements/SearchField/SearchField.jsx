import React from 'react';
import PropTypes from 'prop-types';

const SearchField = ({ input }) => {

  const inputStyles = {
    marginLeft: '-66px'
  }

  return (
    <div className="text-input-container">
      <input
        {...input}
        style={inputStyles}
      />
    </div>
  );
};

SearchField.propTypes = {
  input: PropTypes.shape({
    id: PropTypes.string,
    type: PropTypes.string,
    placeholder: PropTypes.string,
    value: PropTypes.string,
    onChange: PropTypes.func,
  })
};

export default SearchField;
