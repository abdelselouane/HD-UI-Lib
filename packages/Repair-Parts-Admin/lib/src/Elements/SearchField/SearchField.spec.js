import React from 'react';
import SearchField from './SearchField';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as Enzyme from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new ReactSixteenAdapter() });

describe('<SearchField />', () => {
  let wrapper;
  let props;

  afterEach(() => {
    jest.clearAllMocks();
  });

  beforeAll(() => {
    props = {
      input: {
        id: 'searchField',
        type: 'text',
        placeholder: "placeholder",
        value: '123',
        onChange: jest.fn()
      },
      action: {
        id: 'searchLink',
        onClick: jest.fn(),
        text: "Search"
      }
    };
    wrapper = shallow(<SearchField {...props} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call handleOnChange on Search Field Change', () => {
    const spy = jest.spyOn(props.input, 'onChange');
    const searchField = wrapper.find('#searchField');
    searchField.simulate('change');
    expect(spy).toHaveBeenCalled();
  });

  it('should call handleSearch on Search Link Click', () => {
    const spy = jest.spyOn(props.action, 'onClick');
    const searchLink = wrapper.find('#searchLink');
    searchLink.simulate('click');
    expect(spy).toHaveBeenCalled();
  });

});
