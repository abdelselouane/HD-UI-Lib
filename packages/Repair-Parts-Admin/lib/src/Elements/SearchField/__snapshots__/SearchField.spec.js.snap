// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> should render properly 1`] = `
<div
  className="text-input-container"
>
  <input
    id="searchField"
    onChange={[MockFunction]}
    placeholder="placeholder"
    type="text"
    value="123"
  />
  <a
    id="searchLink"
    onClick={[MockFunction]}
    style={
      Object {
        "color": "#FF7849",
        "cursor": "pointer",
        "float": "right",
        "fontSize": "20px",
        "fontWeight": 500,
        "marginLeft": "-80px",
        "marginTop": "27px",
        "padding": "5px",
        "position": "absolute",
        "textDecoration": "none",
      }
    }
  >
    Search
  </a>
</div>
`;
